gk/main.c

```diff
@@ -933,25 +933,29 @@ gk_setup_rss(struct gk_config *gk_conf)
 		gk_queues_back[i] = gk_conf->instances[i].rx_queue_back;
 	}
 
-	ret = gatekeeper_setup_rss(
-		port_front, gk_queues_front, gk_conf->num_lcores);
-	if (ret < 0)
-		goto out;
+	if (gk_conf->net->front.rss) {
```

Comment:
By the way, instead of modifying `gk_setup_rss()` and `gt_setup_rss()`, shouldn't we modify their callers?

Reply:
Even if we are not using RSS on the front interface, we might be using it on the back interface, so the function `gk_setup_rss()` needs to continue to check for that case.

I think we can check whether `gt_setup_rss()` should be invoked from the caller because we only need to check the front interface, unlike `gk_setup_rss()`.